core/src/sample_performance_service.rs

```diff
@@ -51,19 +72,10 @@ impl SamplePerformanceService {
         blockstore: &Arc<Blockstore>,
         exit: Arc<AtomicBool>,
     ) {
-        let (bank, highest_slot) = {
-            let forks = bank_forks.read().unwrap();
-            (forks.root_bank(), forks.highest_slot())
-        };
-
         // Store the absolute transaction counts to that we can compute the
         // difference between these values at points in time to figure out
         // how many transactions occurred in that timespan.
-        let mut snapshot = SamplePerformanceSnapshot {
-            num_transactions: bank.transaction_count(),
-            num_non_vote_transactions: bank.non_vote_transaction_count_since_restart(),
-            highest_slot,
-        };
+        let mut snapshot = Snapshot::from_forks(&bank_forks.read().unwrap());
```

Comment:
```suggestion
        let mut snapshot = SamplePerformanceSnapshot::from_forks(&bank_forks.read().unwrap());
```

Reply:
Sorry for sharing it half baked %)
Renamed `SamplePerformanceSnapshot` to `StatsSnapshot`.
It is already inside `sample_performance_service.rs`, so the `SamplePerformance` part in the name is just “noise”.